oshi-core/src/main/java/oshi/hardware/platform/linux/LinuxUsbDevice.java

```diff
@@ -82,73 +90,76 @@ public LinuxUsbDevice(String name, String vendor, String vendorId, String produc
     }
 
     private static List<UsbDevice> getUsbDevices() {
-        // Enumerate all usb devices and build information maps
-        Udev.UdevContext udev = Udev.INSTANCE.udev_new();
-        // Create a list of the devices in the 'usb' subsystem.
-        UdevEnumerate enumerate = Udev.INSTANCE.udev_enumerate_new(udev);
-        Udev.INSTANCE.udev_enumerate_add_match_subsystem(enumerate, "usb");
-        Udev.INSTANCE.udev_enumerate_scan_devices(enumerate);
-        UdevListEntry devices = Udev.INSTANCE.udev_enumerate_get_list_entry(enumerate);
-
         // Build a list of devices with no parent; these will be the roots
         List<String> usbControllers = new ArrayList<>();
 
-        // Maps to store information using device node path as the key
+        // Maps to store information using device syspath as the key
         Map<String, String> nameMap = new HashMap<>();
         Map<String, String> vendorMap = new HashMap<>();
         Map<String, String> vendorIdMap = new HashMap<>();
         Map<String, String> productIdMap = new HashMap<>();
         Map<String, String> serialMap = new HashMap<>();
         Map<String, List<String>> hubMap = new HashMap<>();
 
-        // For each item enumerated, store information in the maps
-        for (UdevListEntry dev_list_entry = devices; dev_list_entry != null; dev_list_entry = Udev.INSTANCE
-                .udev_list_entry_get_next(dev_list_entry)) {
+        // Enumerate all usb devices and build information maps
+        Udev.UdevContext udev = Udev.INSTANCE.udev_new();
```

Comment:
Looking at this again, I see that it's possible to return null if the native call fails, but [of the four times we call this](https://github.com/oshi/oshi/search?q=udev_new), we only null check one of them.   No need to change this PR for that but I'll have to think about whether we should update all of them.